data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/GrokMatchUtil.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.logstash.mapping;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class GrokMatchUtil {
+    private static final String GROK_MATCH_PATTERN_REGEX = "(%\\{[^{}:]+)(:)?([^{}]+)?(}(\\s+)?)";
+    private static final Pattern GROK_MATCH_PATTERN = Pattern.compile(GROK_MATCH_PATTERN_REGEX);
+
+    public static String convertGrokMatchPattern(final String matchPattern) {
+        Matcher grokMatchPatternMatcher = GROK_MATCH_PATTERN.matcher(matchPattern);
+        StringBuilder convertedGrokMatchPattern = new StringBuilder();
+        while(grokMatchPatternMatcher.find()) {
+            convertedGrokMatchPattern.append(grokMatchPatternMatcher.group(1));
```

Comment:
These are magic numbers.  What do they mean?  We should define constants for them that describe what they mean.  What is in group 1? Group 2?  etc.

Reply:
I'll add group names for regex.